x/nft/exported/nft.go

```diff
@@ -10,7 +10,7 @@ import (
 type NFT interface {
 	GetID() string
 	GetName() string
-	GetOwner() sdk.AccAddress
+	GetOwner() (sdk.AccAddress, error)
```

Comment:
when would `GetOwner` return error?

Reply:
https://github.com/crypto-org-chain/chain-main/pull/516/files#diff-56c6844365f0697399efd4332313e84170925e99bd1479be9a9403dbd3e7d564R37

Reply:
Current code assumes that the bech32 owner address is valid and discards the error when converting bech32 address into `sdk.AccAddress`.

Reply:
I looked down and it seems the error will only be if `SplitKeyOwner` returns an error... which would only be in the case of a corrupted storage? if so, perhaps just panic there instead of bubbling this up (panics should be caught by the Cosmos SDK middleware afaik)?

Reply:
Ok

Reply:
or is this interface ever called on non-validated user input?

Reply:
@yihuang @leejw51crypto any thoughts on this? This would be good to have a second pair of eyes

Reply:
There are two functions named `GetOwner` in the module. This `GetOwner` does not call `SplitKeyOwner`. It only parses bech32 address. Even if we find that this function is never called on non-validated user input, this'll not guarantee future changes to the code will not do that. We can decide between two options:

- Returning `error`
- panic

@yihuang @leejw51crypto @linfeng-crypto @JayT106 any comments?

Reply:
Currently `BaseNFT` is used in the `Collection` type which is part of genesis state. Genesis state I think should be considered non-validated user input? it's supposed to be validated in `ValidateGenesis`.

Reply:
I'd go with how this is done in Cosmos SDK or other modules for consistency

Reply:
agree, for corrupted storage, panic make sense

Reply:
agree to use panic, there is an example for handling the error after calling `AccAddressFromBech32`
https://github.com/cosmos/cosmos-sdk/blob/be4a965599dfae6992451e4b5135c487ed45053b/x/bank/types/balance.go#L17

Reply:
Done